physionet-django/console/test_views.py

```diff
@@ -614,3 +615,159 @@ def test_frontpage_button_delete(self):
+                'is_active': True,
+                'html_content': self.event_agreement_html_content,
+                'access_template': self.event_agreement_access_template
+            })
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'console/event_agreement_detail.html')
+        self.assertContains(response,
+                            'Version may only contain numbers and dots, and must begin and end with a number.')
+
+    def test_delete_event_agreement(self):
+        """tests the view that deletes an event agreement"""
+
+        event_agreement = self.test_add_event_agreement_valid()
+
+        # Delete the event Agreement
+        response = self.client.post(
+            reverse('event_agreement_delete', args=[event_agreement.pk]))
+
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(EventAgreement.objects.filter(slug=self.event_agreement_slug).exists(), False)
+
+    def test_event_agreement_new_version_valid(self):
+        """tests the view that adds a valid new version of event agreement"""
+
+        event_agreement = self.test_add_event_agreement_valid()
+
+        # Create an event Agreement
+        response = self.client.post(
+            reverse('event_agreement_new_version', args=[event_agreement.pk]),
+            data={
+                'name': self.event_agreement_name,
+                'version': self.event_agreement_version_new_version,
+                'slug': self.event_agreement_slug_new_version,
+                'is_active': True,
+                'html_content': self.event_agreement_html_content,
+                'access_template': self.event_agreement_access_template
+            })
+
+        self.assertEqual(response.status_code, 302)
+        event_agreement = EventAgreement.objects.get(slug=self.event_agreement_slug_new_version)
+        self.assertEqual(event_agreement.version, self.event_agreement_version_new_version)
+
+    def test_event_agreement_new_version_invalid(self):
+        """tests the view that adds an invalid new version of event agreement"""
+
+        event_agreement = self.test_add_event_agreement_valid()
+
+        # Create an event Agreement
+        response = self.client.post(
+            reverse('event_agreement_new_version', args=[event_agreement.pk]),
+            data={
+                'name': self.event_agreement_name,
+                'version': self.event_agreement_version_new_version,
+                'slug': self.event_agreement_slug,
+                'is_active': True,
+                'html_content': self.event_agreement_html_content,
+                'access_template': self.event_agreement_access_template
+            })
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'console/event_agreement_new_version.html')
+        self.assertContains(response, 'Event agreement with this Slug already exists.')
```

Comment:
another test it would be good to add:

- try to add a project with a new slug but an existing version
- system raises a 200 and tells the user that the version already exists